packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -130,10 +136,36 @@ describe('HttpServer (integration)', () => {
     server = new HttpServer(dummyRequestHandler);
     await server.start();
     const port = server.port;
-    const anotherServer = new HttpServer(dummyRequestHandler, {port: port});
+    const anotherServer = new HttpServer(dummyRequestHandler, {
+      port: port,
+    } as HttpOptions);
     expect(anotherServer.start()).to.be.rejectedWith(/EADDRINUSE/);
   });
 
+  it('supports HTTPS protocol with key and certificate files', async () => {
+    const httpsServer: HttpServer = givenHttpsServer({});
+    await httpsServer.start();
+    const response = await httpsGetAsync(httpsServer.url);
+    expect(response.statusCode).to.equal(200);
+  });
+
+  it('supports HTTPS protocol with a pfx file', async () => {
+    const httpsServer: HttpServer = givenHttpsServer({usePfx: true});
+    await httpsServer.start();
+    const response = await httpsGetAsync(httpsServer.url);
+    expect(response.statusCode).to.equal(200);
+  });
+
+  it('handles IPv6 loopback address in HTTPS', async () => {
+    const httpsServer: HttpServer = givenHttpsServer({
+      host: '::1',
+    } as HttpsOptions);
```

Comment:
No type cast please, find a way how to express the type constraints in a way that TypeScript compiler understands.

BTW `givenHttpsServer` is not even accepting `HttpsOptions`, it uses a different interface for the options argument.